### PR TITLE
chore: Integrate fomat cmake

### PR DESCRIPTION
**Pull request by sheny1xuan**
+ Clang_format is not well used in our project, I think it's beacause we don't have easy used format script. Ref to [bustub](https://github.com/cmu-db/bustub/blob/master/build_support/run_clang_format.py), integrate clang-format in cmake.
+ We can format code using `make format` after configuring project with cmake.
+ Format current files in `casbin/*`, `include/*` and `tests/*`.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@EmperorYP7

File: casbin/effect/default_effector.cpp [hunk 1]
```diff
@@ -1,55 +1,58 @@
...
 
 /**
- * MergeEffects merges all matching results collected by the enforcer into a single decision.
+ * MergeEffects merges all matching results collected by the enforcer into a
+ * single decision.
  */
-bool DefaultEffector :: MergeEffects(std::string expr, std::vector<Effect> effects, std::vector<float> results) {
+bool DefaultEffector ::MergeEffects(std::string expr,
+                                    std::vector<Effect> effects,
+                                    std::vector<float> results) {
```
Comment (hsluoyz) [inline casbin/effect/default_effector.cpp:33]: Don't split into lines, increase the max row length

File: casbin/internal_api.cpp [hunk 2]
```diff
@@ -43,165 +42,164 @@ bool Enforcer::addPolicy(const std::string& sec, const std::string& p_type, cons
...
+        this->BuildIncrementalRoleLinks(policy_add, p_type, {newRule});
     }
     if (m_watcher && m_auto_notify_watcher) {
-        if(IsInstanceOf<WatcherUpdatable>(m_watcher.get())) {
-            std::dynamic_pointer_cast<WatcherUpdatable>(m_watcher)->UpdateForUpdatePolicy(oldRule, newRule);
-        }
-        else {
+        if (IsInstanceOf<WatcherUpdatable>(m_watcher.get())) {
+            std::dynamic_pointer_cast<WatcherUpdatable>(m_watcher)
+                ->UpdateForUpdatePolicy(oldRule, newRule);
```
Comment (hsluoyz) [inline casbin/internal_api.cpp:177]: Too ugly

File: casbin/ip_parser/parser/equal.cpp [hunk 3]
```diff
@@ -3,17 +3,14 @@
...
 
 namespace casbin {
 
 bool equal(const IPMask& m1, const IPMask& m2) {
-    if(m1.size() != m2.size())
-        return false;
-    for(int i = 0 ; i < m1.size() ; i++) {
-        if(m1[i] != m2[i] )
-            return false;
+    if (m1.size() != m2.size()) return false;
```
Comment (hsluoyz) [inline casbin/ip_parser/parser/equal.cpp:11]: if must have brackets, we don't allow single-line if

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@sheny1xuan change the format

File: casbin/effect/default_effector.cpp (hunk 1 above)
Comment (sheny1xuan) [reply casbin/effect/default_effector.cpp:33]: > @sheny1xuan change the format

It's formated by origin configuration file, I will reformat it latter.

**Comment by sheny1xuan** [comment]
It's formated by origin configuration file, I will reformat it latter.

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**